### PR TITLE
DEV: moves channel-archive-status to channels subscriptions

### PR DESCRIPTION
**Pull request by jjaffeux**
It was quite an oddball because its a global subscription created on each channel. channels manager now allows us to elegantly solve this case.

Future work would rationalize use of message bus channels to limit the number of subscriptions.

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/services/chat-subscriptions-manager.js [hunk 1]
```diff
@@ -64,6 +66,27 @@ export default class ChatSubscriptionsManager extends Service {
...
+      this.messageBus.subscribe(
+        "/chat/channel-archive-status",
+        this._onChannelArchiveStatusUpdate,
+        lastId
+      );
+    }
+  }
+
+  _stopChannelArchiveStatusSubscription() {
+    this._super(...arguments);
```
Comment (tgxworld) [inline plugins/chat/assets/javascripts/discourse/services/chat-subscriptions-manager.js:80]: I don't think we need to call `_super` here.

File: plugins/chat/app/services/chat_publisher.rb [hunk 2]
```diff
@@ -230,6 +230,8 @@ def self.publish_chat_channel_metadata(chat_channel)
     )
   end
 
+  CHANNEL_ARCHIVE_STATUS_BUS_CHANNEL = "/chat/channel-archive-status"
```
Comment (tgxworld) [inline plugins/chat/app/services/chat_publisher.rb:233]:
```suggestion
  CHANNEL_ARCHIVE_STATUS_MESSAGE_BUS_CHANNEL = "/chat/channel-archive-status"
```

For consistency with the other constants.

File: plugins/chat/app/services/chat_publisher.rb [hunk 3]
```diff
@@ -238,7 +240,7 @@ def self.publish_archive_status(
     total_messages:
   )
     MessageBus.publish(
-      "/chat/channel-archive-status",
+      CHANNEL_ARCHIVE_STATUS_BUS_CHANNEL,
```
Comment (tgxworld) [inline plugins/chat/app/services/chat_publisher.rb:243]:
```suggestion
      CHANNEL_ARCHIVE_STATUS_MESSAGE_BUS_CHANNEL,
```

**Review by tgxworld — APPROVED** [review]
Some minor comments but fix looks good to me otherwise.

**Comment by jjaffeux** [comment]
Thanks @tgxworld :+1:

**Pull request merged**